src/modules/qt/filter_qresize.cpp

```diff
@@ -0,0 +1,344 @@
+	{
+#if QT_VERSION >= 0x050200
+		QImage src(iwidth, owidth, QImage::Format_RGBA8888);
+		convert_mlt_to_qimage_rgba( input, &src, iwidth, iheight );
+		QImage dest( output, owidth, oheight, QImage::Format_RGBA8888 );
+		dest.fill(Qt::transparent);
+		QPainter painter(&dest);
+		painter.drawImage(QPoint((owidth - iwidth) / 2, (oheight - iheight) / 2), src);
+		painter.end();
+		convert_qimage_to_mlt_rgba( &dest, output, owidth, oheight );
+		return;
+#endif
+		while ( size -- )
+		{
+			*p ++ = 0;
+			*p ++ = 0;
+			*p ++ = 0;
+			*p ++ = alpha_value;
+		}
+	}
+	else if ( bpp == 2 )
+	{
+		while( size -- )
+		{
+			*p ++ = 16;
+			*p ++ = 128;
+		}
+		offset_x -= offset_x % 4;
+	}
+	else
+	{
+		size *= bpp;
+		while ( size-- )
+			*p ++ = 0;
+	}
+
+	out_line = output + offset_y * ostride;
+	out_line += offset_x;
+
+	// Loop for the entirety of our output height.
+	while ( iheight -- )
+	{
+		// We're in the input range for this row.
+		memcpy( out_line, in_line, istride );
+
+		// Move to next input line
+		in_line += istride;
+
+		// Move to next output line
+		out_line += ostride;
+	}
+}
+
+/** A padding function for frames - this does not rescale, but simply
+	resizes.
+*/
+
+static uint8_t *frame_resize_image( mlt_frame frame, int owidth, int oheight, mlt_image_format format )
+{
+	// Get properties
+	mlt_properties properties = MLT_FRAME_PROPERTIES( frame );
+
+	// Get the input image, width and height
+	uint8_t *input = (uint8_t *)mlt_properties_get_data( properties, "image", NULL );
+	uint8_t *alpha = mlt_frame_get_alpha( frame );
+	int alpha_size = 0;
+	mlt_properties_get_data( properties, "alpha", &alpha_size );
+	int bpp = 0;
+	mlt_image_format_size( format, owidth, oheight, &bpp );
+
+	int iwidth = mlt_properties_get_int( properties, "width" );
+	int iheight = mlt_properties_get_int( properties, "height" );
+
+	// If width and height are correct, don't do anything
+	if ( iwidth < owidth || iheight < oheight )
+	{
+		uint8_t alpha_value = mlt_properties_get_int( properties, "resize_alpha" );
+		// Create the output image
+		uint8_t *output = (uint8_t*)mlt_pool_alloc( owidth * ( oheight + 1 ) * bpp );
+
+		// Call the generic resize
+		resize_image( output, owidth, oheight, input, iwidth, iheight, bpp, format, alpha_value );
+
+		// Now update the frame
+		mlt_frame_set_image( frame, output, owidth * ( oheight + 1 ) * bpp, mlt_pool_release );
+
+		// We should resize the alpha too
+		if ( format != mlt_image_rgb24a && alpha && alpha_size >= iwidth * iheight )
+		{
+			alpha = resize_alpha( alpha, owidth, oheight, iwidth, iheight, alpha_value );
+			if ( alpha )
+				mlt_frame_set_alpha( frame, alpha, owidth * oheight, mlt_pool_release );
+		}
+
+		// Return the output
+		return output;
+	}
+	// No change, return input
+	return input;
+}
+
+/** Do it :-).
+*/
+
+static int filter_get_image( mlt_frame frame, uint8_t **image, mlt_image_format *format, int *width, int *height, int writable )
+{
+	int error = 0;
```

Comment:
Instead of duplicating so much code from filter_resize, could you detect in this function if QImage can be used based on the format? If QImage can be used, then use it. If not, then "fall back" to an encapsulated instance of filter_resize.